mlflow/models/evaluation/utils/trace.py

```diff
@@ -0,0 +1,117 @@
+    global_config_snapshot = AUTOLOGGING_INTEGRATIONS.copy()
+
+    for flavor in FLAVOR_TO_MODULE_NAME:
+        # TODO: Remove this once Gluon autologging is actually removed
+        if flavor == "gluon":
+            continue
+
+        try:
+            if autolog := _get_autolog_function(flavor):
+                original_config = global_config_snapshot.get(flavor, {}).copy()
+
+                # If autologging is explicitly disabled, do nothing.
+                if original_config.get("disable", False):
+                    continue
+
+                elif enable_tracing and _is_trace_autologging_supported(flavor):
+                    # set all log_xyz params to False except log_traces
+                    new_config = {
+                        k: False if k.startswith("log_") else v for k, v in original_config.items()
+                    }
+                    new_config |= {"log_traces": True, "silent": True}
+                    _kwargs_safe_invoke(autolog, new_config)
+                    trace_enabled_flavors.append(flavor)
+                elif flavor in AUTOLOGGING_INTEGRATIONS:
+                    # For flavors that does not support tracing, disable autologging
+                    autolog(disable=True)
+
+                flavor_to_original_config[flavor] = original_config
+
+        except Exception as e:
+            if isinstance(e, ImportError):
+                _logger.debug(f"Failed to import {flavor}. Skip updating autologging.")
+            else:
+                _logger.info(f"Failed to update autologging configuration for flavor {flavor}. {e}")
+
+            # Autologging configuration might be updated before the exception is raised,
+            # which needs to be reverted.
+            AUTOLOGGING_INTEGRATIONS.pop(flavor, None)
+
+    if trace_enabled_flavors:
+        _logger.info(
+            "Tracing is temporarily enabled during the model evaluation for computing some "
+            "metrics and debugging. To disable tracing, call `mlflow.autolog(disable=True)`."
+        )
+
+    try:
+        yield
+    finally:
+        # Restore original autologging configurations.
+        for flavor, original_config in flavor_to_original_config.items():
+            autolog = _get_autolog_function(flavor)
+            try:
+                if original_config:
+                    _kwargs_safe_invoke(autolog, original_config)
+                    AUTOLOGGING_INTEGRATIONS[flavor] = original_config
+                else:
+                    # If the original configuration is empty, autologging was not enabled before
+                    autolog(disable=True)
+                    # We also need to remove the configuration entry from AUTOLOGGING_INTEGRATIONS,
+                    # so as not to confuse with the case user explicitly disabled autologging.
+                    AUTOLOGGING_INTEGRATIONS.pop(flavor, None)
+            except ImportError:
```

Comment:
What can `ImportError`? `autolog`?

Reply:
Yes, there is an edge case that `autolog()` does not raise an import error but `autolog(disable=True)` does.